parts/k8s/addons/kubernetesmasteraddons-calico-daemonset.yaml

```diff
@@ -190,222 +93,86 @@ spec:
           # connections.
-          - name: TYPHA_CONNECTIONREBALANCINGMODE
-            value: "kubernetes"
-          - name: TYPHA_DATASTORETYPE
-            value: "kubernetes"
-          - name: TYPHA_HEALTHENABLED
-            value: "true"
+        - name: TYPHA_CONNECTIONREBALANCINGMODE
+          value: "kubernetes"
+        - name: TYPHA_DATASTORETYPE
+          value: "kubernetes"
+        - name: TYPHA_HEALTHENABLED
+          value: "true"
           # Uncomment these lines to enable prometheus metrics.  Since Typha is host-networked,
           # this opens a port on the host, which may need to be secured.
           #- name: TYPHA_PROMETHEUSMETRICSENABLED
           #  value: "true"
           #- name: TYPHA_PROMETHEUSMETRICSPORT
           #  value: "9093"
         livenessProbe:
-          httpGet:
-            path: /liveness
-            port: 9098
-            host: localhost
+          exec:
+            command:
+            - calico-typha
+            - check
+            - liveness
           periodSeconds: 30
           initialDelaySeconds: 30
         readinessProbe:
-          httpGet:
-            path: /readiness
-            port: 9098
-            host: localhost
+          exec:
+            command:
+            - calico-typha
+            - check
+            - readiness
           periodSeconds: 10
-
----
-
-# Typha Horizontal Autoscaler Cluster Role
-apiVersion: rbac.authorization.k8s.io/v1
-kind: ClusterRole
-metadata:
-  name: typha-cpha
-  labels:
-    kubernetes.io/cluster-service: "true"
-    addonmanager.kubernetes.io/mode: "EnsureExists"
-rules:
-  - apiGroups: [""]
-    resources: ["nodes"]
-    verbs: ["list"]
-
 ---
 
-# Typha Horizontal Autoscaler Cluster Role Binding
-apiVersion: rbac.authorization.k8s.io/v1
-kind: ClusterRoleBinding
-metadata:
-  name: typha-cpha
-  labels:
-    kubernetes.io/cluster-service: "true"
-    addonmanager.kubernetes.io/mode: "EnsureExists"
-roleRef:
-  apiGroup: rbac.authorization.k8s.io
-  kind: ClusterRole
-  name: typha-cpha
-subjects:
-  - kind: ServiceAccount
-    name: typha-cpha
-    namespace: kube-system
-
----
-
-# Typha Horizontal Autoscaler ConfigMap
-kind: ConfigMap
-apiVersion: v1
-metadata:
-  name: calico-typha-horizontal-autoscaler
-  namespace: kube-system
-  labels:
-    kubernetes.io/cluster-service: "true"
-    addonmanager.kubernetes.io/mode: "EnsureExists"
-data:
-  ladder: |-
-    {
-      "coresToReplicas": [],
-      "nodesToReplicas":
-      [
-        [1, 1],
-        [10, 2],
-        [100, 3],
-        [250, 4],
-        [500, 5],
-        [1000, 6],
-        [1500, 7],
-        [2000, 8]
-      ]
-    }
-
----
+# This manifest creates a Pod Disruption Budget for Typha to allow K8s Cluster Autoscaler to evict
 
-# Typha Horizontal Autoscaler Deployment
-apiVersion: extensions/v1beta1
-kind: Deployment
+apiVersion: policy/v1beta1
+kind: PodDisruptionBudget
```

Comment:
You should not include the PodDisruptionBudget object. It was added at a time but was removed cause it caused problem with the upgrade problem (see https://github.com/Azure/aks-engine/pull/300).

Also a PodDisruptionBudget can not be updated at the moment. It needs to be removed and recreated if you want to change the spec so it's better to not include it.